examples/apps/image-folder-support/app.py

```diff
@@ -0,0 +1,256 @@
+            label=f"Total {len(image_paths)} images",
+            images=image_paths,
+            key="preview_input_images",
+            captions=[f"{p.name}" for p in image_paths],
+            use_container_width=False,
+        )
+        img_path_str = str(img_path)
+        st.image(img_path_str, caption=img_path_str)
+
+    st.divider()
+    st.subheader("Run inferences")
+    request_to_run_inference = st.button(
+        "Run inferences",
+        on_click=bulk_inference,
+        kwargs={
+            "image_paths": image_paths,
+            "image_folder_path": local_image_folder,
+        },
+        type="secondary",
+    )
+    result = st.session_state.get("result", None)
+    if result:
+        img_with_preds = [
+            str(pred.img_with_preds_path) for pred in result.image_predictions
+        ]
+        with st.expander("Preview prediction results"):
+            img_path = image_select(
+                label=f"Total {len(image_paths)} images",
+                images=img_with_preds,
+                key="preview_pred_images",
+                captions=[f"{p.name}" for p in image_paths],
+                use_container_width=False,
+            )
+            img_path_str = str(img_path)
+            st.image(img_path_str, caption=img_path_str)
+
+            for res in result.image_predictions:
+                if str(res.img_with_preds_path) == img_path_str:
+                    preds = res.predictions
+                    if not preds or not isinstance(preds[0], SegmentationPrediction):
+                        continue
+                    coverage = segmentation_class_pixel_coverage(preds).values()
+                    labels = [val[1] for val in coverage]
+                    sizes = [val[0] for val in coverage]
+                    unclassified_pixels = 1 - sum(sizes)
+                    sizes.append(unclassified_pixels)
+                    labels.append("unclassified")
+                    fig = px.pie(
+                        values=sizes,
+                        names=labels,
+                        title="Predicted pixel distribution over a single image",
+                    )
+                    st.plotly_chart(fig)
+
+        all_preds = []
+        for pred in result.image_predictions:
+            all_preds.extend(pred.predictions)
+
+        if not all_preds:
+            st.warning("No predictions found.")
+            st.stop()
+
+        if isinstance(all_preds[0], SegmentationPrediction):
+            coverage = segmentation_class_pixel_coverage(all_preds).values()
+            labels = [val[1] for val in coverage]
+            sizes = [val[0] for val in coverage]
+            unclassified_pixels = 1 - sum(sizes)
+            sizes.append(unclassified_pixels)
+            labels.append("unclassified")
+            fig = px.pie(
+                values=sizes,
+                names=labels,
+                title="Predicted pixel distribution over all images",
+            )
```

Comment:
I we need to add something like Describe() to the pipeline to generate useful and common stats like defect distribution and defect coverage

Reply:
yeah, we can consider supporting plotting and data visualization.
though, i'd wait to see a few more user stories and decide what kind of visualization library are mostly suitable for them. Here I just picked one that I'm familiar with, which may not be the best one.